packages/core/src/kernelCoreTemplate.js

```diff
@@ -82,6 +90,43 @@
     Object.freeze(kernel)
     return kernel
 
+    function scuttleGlobalThis(extraAvoids = new Array()) {
+      let props = [...Object.getOwnPropertyNames(globalThis)]
+      if (globalThis?.Window?.prototype) {
+        props = [...props, ...Object.getOwnPropertyNames(Window.prototype)]
+      }
+      if (globalThis?.EventTarget?.prototype) {
+        props = [...props, ...Object.getOwnPropertyNames(EventTarget.prototype)]
+      }
+
+      const avoids = [
+        // non configurables
+        'location', 'top', 'global', 'self', 'window', 'Infinity', 'NaN', 'document',
+        // support LM,SES exported APIs
+        'LavaPack', 'Compartment',
+        // support polyfill
+        'globalThis',
+      ]
+
+      for (const prop of props) {
+        if (!avoids.includes(prop) && !extraAvoids.includes(prop)) {
+          // these props can't have getters, use undefined value instead
+          const desc = ['undefined', 'chrome', 'constructor'].includes(prop) ?
+            { value: undefined } :
+            {
+              set: () => {},
+              get: () => {
+                throw new Error(
+                  `LavaMoat - property "${prop}" of globalThis is inaccessible under scuttling mode. ` +
+                  `To learn more visit https://github.com/LavaMoat/LavaMoat/pull/360.`)
+              },
+            }
+          Object.defineProperty(window, prop, desc)
```

Comment:
I know it probably gets frozen later, but it'd be nice to make these non-configurable

Reply:
nice, this is better than just removing them